libs/base/System/Info.idr

```diff
@@ -14,3 +14,11 @@ codegen = prim__codegen
 export
 isWindows : Bool
 isWindows = os `elem` ["windows", "mingw32", "cygwin32"]
+
+%foreign "C:idris2_getNProcessors, libidris2_support"
+prim__getNProcessors : PrimIO Int
+
+export
+getNProcessors : IO Int
```

Comment:
Should we make this a `Nat`?

Reply:
I don't think so, since the C functions can return -1 to indicate an error.
We should maybe do something clever with error reporting (e.g. `Either`), but it's difficult due to the different ways Linux, Mac, and Windows do readable error messages and error codes.

Reply:
Then a `Maybe Nat`? Basically a safe wrapper for the primitive.